homeassistant/components/ring/binary_sensor.py

```diff
@@ -53,70 +71,92 @@ async def async_setup_entry(
-        description: RingBinarySensorEntityDescription,
+        device: RingDeviceT,
+        coordinator: RingListenCoordinator,
+        description: RingBinarySensorEntityDescription[RingDeviceT],
     ) -> None:
-        """Initialize a sensor for Ring device."""
+        """Initialize a binary sensor for Ring device."""
         super().__init__(
             device,
             coordinator,
         )
         self.entity_description = description
-        self._ring = ring
         self._attr_unique_id = f"{device.id}-{description.key}"
-        self._update_alert()
+        self._attr_entity_registry_enabled_default = (
+            description.entity_registry_enabled_default
+        )
+        self._attr_is_on = False
+        self._active_alert: RingEvent | None = None
+        self._cancel_callback: TimerHandle | None = None
 
     @callback
-    def _handle_coordinator_update(self, _: Any = None) -> None:
-        """Call update method."""
-        self._update_alert()
-        super()._handle_coordinator_update()
+    def _async_handle_event(self, alert: RingEvent) -> None:
+        """Handle the event."""
+        self._attr_is_on = True
+        self._active_alert = alert
+        loop = self.hass.loop
+        when = loop.time() + alert.expires_in
+        if self._cancel_callback:
+            self._cancel_callback.cancel()
+        self._cancel_callback = loop.call_at(when, self._async_cancel_event)
 
     @callback
-    def _update_alert(self) -> None:
-        """Update active alert."""
-        self._active_alert = next(
-            (
-                alert
-                for alert in self._ring.active_alerts()
-                if alert["kind"] == self.entity_description.key
-                and alert["doorbot_id"] == self._device.id
-            ),
-            None,
+    def _async_cancel_event(self) -> None:
+        """Clear the event."""
+        self._cancel_callback = None
+        self._attr_is_on = False
+        self._active_alert = None
+        self.async_write_ha_state()
+
+    def _get_coordinator_alert(self) -> RingEvent | None:
+        alerts = (
+            alert
+            for alert in self.coordinator.ring_api.active_alerts()
+            if alert.kind == self.entity_description.key
+            and alert.doorbot_id == self._device.device_api_id
         )
+        return next(alerts, None)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if alert := self._get_coordinator_alert():
+            self._async_handle_event(alert)
+        super()._handle_coordinator_update()
 
     @property
-    def is_on(self) -> bool:
-        """Return True if the binary sensor is on."""
-        return self._active_alert is not None
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self.coordinator.event_listener.started
+
+    async def async_update(self) -> None:
+        """All updates are passive."""
```

Comment:
You don't have to define this, rather set `_attr_should_poll = False`

Reply:
`async_update` is an abstract method on the `BaseCoordinatorEntity` so this needs to be defined.